src/capi/object.cpp

```diff
@@ -614,6 +614,40 @@ extern "C" PyObject** _PyObject_GetDictPtr(PyObject* obj) noexcept {
     }
 }
 
+// This function returns the copy of object's dict, not a writable dict.
+extern "C" PyObject* PyObject_GetDict(PyObject* obj) noexcept {
+    PyTypeObject* tp = Py_TYPE(obj);
+    if (!tp->instancesHaveHCAttrs()) {
+        if (!tp->instancesHaveDictAttrs())
+            return NULL;
+        return obj->getDict();
```

Comment:
thanks for changing the name but I think this should call `PyDict_Copy(obj->getDict())` because otherwise it's not always a copy as the name suggests...